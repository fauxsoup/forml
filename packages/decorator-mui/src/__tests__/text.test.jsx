import Text from '../text';
import Context from '@forml/context';
import React from 'react';
import { render } from '@testing-library/react';
import * as decorator from '../';

describe('renders', function () {
    let form;
    let title = 'title';
    let description = 'description';

    beforeEach(function () {
        form = { type: 'text' };
    });

    describe('with form options', function () {
        let fields = {
            variant: [undefined, 'h1', 'h2'],
            align: [undefined, 'left', 'right'],
            color: [undefined, 'primary', 'secondary'],
            noWrap: [undefined, true, false],
            paragraph: [undefined, true, false],
            otherProps: [undefined, {}],
        };

        Object.keys(fields).forEach(function (field) {
            fields[field].forEach(function (value) {
                describe(`${field}`, function () {
                    test(`${value}`, function () {
                        if (value !== undefined)
                            form = { ...form, [field]: value };
                        const { container } = render(
                            <Context.Provider value={{ decorator }}>
                                <Text
                                    form={form}
                                    title={title}
                                    description={description}
                                />
                            </Context.Provider>
                        );

                        expect(container).toMatchSnapshot();
                    });
                });
            });
        });
    });
});
