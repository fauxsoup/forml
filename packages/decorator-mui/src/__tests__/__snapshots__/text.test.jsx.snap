// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders with form options align left 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 MuiTypography-alignLeft css-zpphrz-MuiTypography-root"
  />
</div>
`;

exports[`renders with form options align right 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 MuiTypography-alignRight css-hlnzui-MuiTypography-root"
  />
</div>
`;

exports[`renders with form options align undefined 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  />
</div>
`;

exports[`renders with form options color primary 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-vkc4v8-MuiTypography-root"
  />
</div>
`;

exports[`renders with form options color secondary 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-18brw8f-MuiTypography-root"
  />
</div>
`;

exports[`renders with form options color undefined 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  />
</div>
`;

exports[`renders with form options noWrap false 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  />
</div>
`;

exports[`renders with form options noWrap true 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 MuiTypography-noWrap css-fvtpii-MuiTypography-root"
  />
</div>
`;

exports[`renders with form options noWrap undefined 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  />
</div>
`;

exports[`renders with form options otherProps [object Object] 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  />
</div>
`;

exports[`renders with form options otherProps undefined 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  />
</div>
`;

exports[`renders with form options paragraph false 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  />
</div>
`;

exports[`renders with form options paragraph true 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 MuiTypography-paragraph css-z2eky3-MuiTypography-root"
  />
</div>
`;

exports[`renders with form options paragraph undefined 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  />
</div>
`;

exports[`renders with form options variant h1 1`] = `
<div>
  <h1
    class="MuiTypography-root MuiTypography-h1 css-o2w69a-MuiTypography-root"
  />
</div>
`;

exports[`renders with form options variant h2 1`] = `
<div>
  <h2
    class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
  />
</div>
`;

exports[`renders with form options variant undefined 1`] = `
<div>
  <p
    class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
  />
</div>
`;
